cdap-common/src/test/java/io/cdap/cdap/common/service/RetryOnStartFailureServiceTest.java

```diff
@@ -112,20 +112,20 @@ public Boolean call() throws Exception {
 
   @Test
   public void testLoggingContext() {
-    // This test logging context set before the service started is propagated into the service
+    // This test logging context set before the service started is not propagated into the child thread
     final Map<String, String> context = Collections.singletonMap("key", "value");
 
     // Create the service before setting the context.
     Service service = new RetryOnStartFailureService(() -> new AbstractIdleService() {
 
       @Override
       protected void startUp() throws Exception {
-        Assert.assertEquals(context, MDC.getCopyOfContextMap());
+        Assert.assertNull(MDC.getCopyOfContextMap());
```

Comment:
Changing this to check for null makes this basically a useless test. The test was added for a reason, did you check the commit that it was added in and see if there is a jira mentioned? There are probably places where we need to now explicitly set the MDC instead of relying on it being inherited.

Reply:
Yes, I agree. The test now seems pointless. I have made a temporary change just to get other things going. Regarding the places where we need to explicitly set the MDC, I was unable to find those exact places for our project despite debugging a lot. 

Logback Commit that implemented this change: https://github.com/qos-ch/logback/commit/aa7d584ecdb1638bfc4c7223f4a5ff92d5ee6273#

Relevant Jiras:
https://jira.qos.ch/browse/LOGBACK-992
https://jira.qos.ch/browse/LOGBACK-173
https://jira.qos.ch/browse/LOGBACK-920

Reply:
In a broader sense, it looks like MDC.put() is only used in a few places to set a key, do something, then immediately reset the key, so we don't need to worry about usages of the put method. 

MDC.setContextMap() is only called by LoggingContextAccessor.setLoggingContext, which stores the LoggingContext in an InheritableThreadLocal and sets system tags in the MDC. The LoggingContext is only fetched from LoggingContextAccessor by the LogAppender. So it seems to me that we should be ok if we tweak the logic in the LogAppender to initialize a map with the system tags, add the event's MDC on top, then add all the other extra tags on top of that. Though again, this should be verified by checking before and after your change.

Reply:
Added the system tags in LogAppender, verified the app-fabric logs have mdc tags after the change. Added some debug logs in `RetryOnFailureService` and verified the mdc tags were propogated.

Reply:
Removed this test since it is no longer meaningful.